### PR TITLE
A single test, trying to run it on Windows.

### PR DESCRIPTION
**Pull request by dkorolev**
The remaining compilation problem is `opendir/readdir/closedir` and related. I'll solve it in the morning.

@sompylasar : I'll handle the merge between this PR and your Mac OS changes. Btw, asked a friend for a few minutes of attention, and Bricks do build on Mac from your PR after a few changes; there are more issues though, which I'd have to look into.

@deathbaba @sompylasar PTAL.

Thanks,
Dima

### CODE REVIEW

**Comment by deathbaba** [comment]
You can check Alohalytics' file.h - I've already done it.

**Comment by dkorolev** [comment]
Aww, cute, thank you. As soon as I wake up ...

**Comment by dkorolev** [comment]
Tests pass on Windows!

@deathbaba : FYI, I've fixed two bugs that may relate to your code as well.

1. ` ::FindFirstFileA` needs `directory + "\\*.*"`, not just `directory` as the 1st parameter.
2. Threads should be the last member variables of respective classes, otherwise Visual Studio (correctly!) initialized them before the other variables are initialized/nullified.

@deathbaba @sompylasar : PTAL.

**Comment by msft-dkorolev** [comment]
image

@deathbaba @sompylasar @dkorolev : PTAL :)

File: zzz_full_test/everything.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+#include "dflags/dflags.h"
+#include "3party/gtest/gtest-main-with-dflags.h"
+
+#include "./dflags/test.cc"
+#include "./time/test.cc"
+#include "./graph/test.cc"
+#include "./util/test.cc"
+#include "./mq/inmemory/test.cc"
+#include "./net/api/docu/client/docu_02httpclient_02_test.cc"
```
Comment (dkorolev) [inline zzz_full_test/everything.cc]: Wow, this order seems messed up.

File: file/file.h [hunk 2]
```diff
@@ -53,8 +53,12 @@ SOFTWARE.
...
 // Platform-indepenent, injection-friendly filesystem wrapper.
-// TODO(dkorolev): Move the above methods under FileSystem.
 struct FileSystem {
+#ifndef BRICKS_WINDOWS
+  static constexpr char PathSeparatingSlash = '/';
+#else
+  static constexpr char PathSeparatingSlash = '\\';
+#endif
   static inline std::string GetFileExtension(const std::string& file_name) {
     const size_t i = file_name.find_last_of("/\\.");
```
Comment (sompylasar) [inline file/file.h]: This string should likely use the platform-dependent `PathSeparatingSlash` instead of both. Don't forget to add the dot to it.

Comment (dkorolev) [inline file/file.h]: I thought of it and prefer to keep it this way. There's a nonzero chance someone might use `GetFileExtension` on a URL, not to mention that using various slashes as non-slashes as part of the file name (and extension!) is something I'd be glad to relentlessly punish a developer for.

Comment (sompylasar) [inline file/file.h]: Agreed.

File: port.h [hunk 3]
```diff
@@ -90,4 +90,19 @@ SOFTWARE.
...
 
+#ifdef BRICKS_WINDOWS
+
+// #include <windows.h>  -- `#include <windows.h>` does not play well with WSA -- D.K.
+
+// These two headers seem to work fine for both WSA and `FindFirst*` files.
+#include <WS2tcpip.h>
+#include <corecrt_io.h>
+
+// Visual Studio does not define `M_PI`.
```
Comment (sompylasar) [inline port.h]: Should a compiler define a math constant? I thought some math-related headers should define it.

Comment (dkorolev) [inline port.h]: It's more of me being lazy to not rely on a non-standard `M_PI` in the demo. Added issue https://github.com/KnowSheet/Bricks/issues/92

**Comment by dkorolev** [comment]
@deathbaba @sompylasar : Polite ping?

**Comment by sompylasar** [comment]
I've got nothing more to add for now to this discussion.

File: file/test.cc [hunk 4]
```diff
@@ -186,17 +235,23 @@ TEST(File, ScanDir) {
...
     ASSERT_THROW(FileSystem::ScanDir(dir, scanner_before), PathIsNotADirectoryException);
     ASSERT_THROW(FileSystem::ScanDirUntil(dir, scanner_before), PathIsNotADirectoryException);
   }
+#endif
 
   FileSystem::MkDir(dir);
   FileSystem::WriteStringToFile("foo", fn1.c_str());
   FileSystem::WriteStringToFile("bar", fn2.c_str());
+#ifndef BRICKS_WINDOWS
+  // Windows is not friendly with subdirectories.
```
Comment (deathbaba) [inline file/test.cc]: What does it mean? JoinPath should work, MkDir too.

**Pull request merged**